secure-app-model/keyvault/partnerconsent/src/main/java/com/microsoft/store/samples/partnerconsent/security/KeyVaultProvider.java

```diff
@@ -63,89 +53,24 @@ public String getSecret(String secretName)
-     * 
-     * @throws ExecutionException {@link ExecutionException}
-     * @throws InterruptedException {@link InterruptedException}
-     * @throws MalformedURLException {@link MalformedURLException}
-     */
-    private AuthenticationResult getAccessToken(String authorization, String resource, String clientId, String clientSecret) 
-        throws ExecutionException, InterruptedException, MalformedURLException
-    {
-        AuthenticationContext authContext; 
-        AuthenticationResult authResult;
-        ExecutorService service = null;
-        Future<AuthenticationResult> future;
-
-        try
-        {
-            service = Executors.newFixedThreadPool(1);
-            authContext = new AuthenticationContext(authorization, true, service);
-
-            future = authContext.acquireToken(
-                resource,
-                new ClientCredential(
-                    clientId, 
-                    clientSecret), 
-                null);
-            
-            authResult = future.get();
-
-            return authResult;
-        }
-        finally
-        {
-            service.shutdown();
-        }
-    }
 
     /**
      * Gets a client that is capable of interacting with the Azure Key Vault service.
-     *
-     * @param clientId The identifier of the client requesting the token.
-     * @param clientSecret The secure secret of the client requesting the token. 
      * 
      * @return A client that is capable of interacting with the Azure Key Vault service.
      */
-    private KeyVaultClientCustom getKeyVaultClient(String clientId, String clientSecret)
+    private SecretClient getKeyVaultClient()
     {
-        return new KeyVaultClient(new KeyVaultCredentials() 
-        {
-            /**
-             * @param authorization Address of the authority to issue the token.
-             * @param resource Identifier of the target resource that is the recipient of the requested token, a URL.
-             * @param scope The scope of the authentication request.
-             *
-             * @return Access token to be used with Azure Key Vault operations.
-             */
-            @Override
-             public String doAuthenticate(String authorization, String resource, String scope) 
-             {
-                 AuthenticationResult authResult; 
-                 
-                 try 
-                 {
-                     authResult = getAccessToken(authorization, resource, clientId, clientSecret);
- 
-                     return authResult.getAccessToken();
-                 }
-                 catch(Exception ex)
-                 {
-                     ex.printStackTrace();
-                 }
- 
-                 return "";
-             }
-        });
+        DefaultAzureCredential defaultAzureCredential = new DefaultAzureCredentialBuilder().build();
+
+        client = new SecretClientBuilder()
+            .vaultUrl(vaultBaseUrl)
+            .credential(defaultAzureCredential)
```

Comment:
This can be made fluent, by instantiating the DAC credential on line 67 in line here.